extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -86,8 +96,31 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm) thro
     protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, RunnableWithException afterEndTransaction)
             throws Exception {
 
-        tm.begin();
-        Transaction tx = tm.getTransaction();
+        TransactionConfiguration configAnnotation = getTransactionConfiguration(ic);
+        int currentTmTimeout = ((TransactionManagerImple) com.arjuna.ats.jta.TransactionManager.transactionManager())
+                .getTimeout();
+        if (configAnnotation != null && configAnnotation.timeout() != TransactionConfiguration.UNSET_TIMEOUT) {
+            tm.setTransactionTimeout(configAnnotation.timeout());
+        }
+        Transaction tx;
+        try {
+            tm.begin();
+            tx = tm.getTransaction();
+            if (tx.getStatus() == Status.STATUS_NO_TRANSACTION && configAnnotation != null) {
+                //it's a development error to use the @TransactionConfiguration annotation in a method that
+                //is not the entry point of a transaction
+                throw new RuntimeException(
```

Comment:
I'd go for `TransactionalException`

Reply:
Cannot use `TransactionalException` here as it needs a cause exception. In fact it's more a `ConfigurationException` that should occurs during the developement. I don't know if there is such exception already existing in Quarkus that I can use.
I'll stick to RuntimeException for now.